torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -856,9 +1009,62 @@ void SegmentCandidateFinder::findSegments() {
     mergeNodes();
   }
 
+  finalMerge();
+
   finalize();
 }
 
+void SegmentCandidateFinder::finalMerge() {
```

Comment:
Can you please comment here and describe why this `finalMerge` is necessary? A simple example case would be great.

Reply:
Thanks for pointing this out. Added comments in fusion_segmenter.h and a motivating simpler example as a test `FusionDAGMerging_CUDA`.

Reply:
While the example is quite synthetical, I do expect we hit quite a few similar cases where the algorithm on the paper cannot propose a valid merge. One of the main cause I observed now is that we have additional constraints on what kinds of nodes can merge and the original algorithm is a hierarchical/iterative one. So it can be terminated early by our constraints.